core/templates/map/map.html

```diff
@@ -32,6 +42,34 @@
 
         const eventDivs = document.querySelectorAll(`div.event[data-date="${date}"][data-venue-lnglat]`);
 
+        // create a dictionary of all of our venues and their lnglats
+        const lnglatsByVenueId = {}
+        Array.from(eventDivs).filter((obj, index, arr) =>  // filter to unique venue ids
+            index === arr.findIndex(obj2 => obj.getAttribute('data-venue-id') === obj2.getAttribute('data-venue-id'))
+          ).forEach(eventDiv => {
+            const venueId = eventDiv.getAttribute('data-venue-id');
+            lnglatsByVenueId[venueId] = eventDiv.getAttribute('data-venue-lnglat');
+          });
+
+        // each element is an array of venue IDs within ~10m of each other
+        const mergedVenues = [];
+        Object.entries(lnglatsByVenueId).forEach(([venueId, venueLngLat]) => {
+          let foundMatchingVenue = false;
+          mergedVenues.forEach((venueList, index) => {
+            if (areWithinTenMeters(venueLngLat, lnglatsByVenueId[venueList[0]])) {
+              mergedVenues[index].push(venueId);
+              foundMatchingVenue = true;
+            }
+          });
+
+          if (!foundMatchingVenue) {
+            mergedVenues.push([venueId]);
+          }
+        });
+        // convert this array or arrays to an array of strings
+        // these will become our venueIds for the map & eventsByVenueId object
+        const mergedVenueIds = mergedVenues.map(el => el.join(','));
```

Comment:
I'm doing these calculations once a map link has been clicked.

Theoretically, these good be pre-calculated somewhere for all venues, but not sure where? Let me know if this is good for you, or if you have a preference somewhere else.

Reply:
yeah, would make sense to pre-calc them -- in a separate func just like you did `areWithinTenMeters` ..? ie before calling `dateLinks.forEach`, go through all events on the page, collect the venues, and 'cluster' them like you did?

totally ok to not do this right now, but yeah. could be cool! but this is super super super super super great for now......!!!